src/main/java/com/palehorsestudios/alone/player/Player.java

```diff
@@ -427,24 +427,119 @@ public Result gatherFirewood() {
+    resultBuilder.calories(-caloriesBurned);
+    updateWeight(-caloriesBurned);
+    int addedWater;
+    int finalAddedWater = 0;
+    if (successRate == SuccessRate.LOW) {
+      addedWater = 1;
+      updateMorale(1);
+      resultBuilder.morale(1);
+    }
+    else if (successRate == SuccessRate.MEDIUM) {
+      addedWater = 2;
+      updateMorale(1);
+      resultBuilder.morale(2);
+    }
+    else {
+      addedWater = 5;
+      updateMorale(3);
+      resultBuilder.morale(3);
+    }
+    finalAddedWater = this.shelter.updateWater(addedWater);
+    resultBuilder
+        .water(finalAddedWater)
+        .message("You added " + finalAddedWater + " in the water tank.");
+    return resultBuilder.build();
   }
-
+  /**
+   * Activity method for Player to boost morale.
+   *
+   * @return Result of boosting morale attempt.
+   */
   public Result boostMorale() {
-    return null;
+    Result.Builder resultBuilder = new Result.Builder();
+    Set<Item> moraleBoostItems = new HashSet<>();
+    List<Item> moraleBoostItemsOwn = new ArrayList<>();
+    moraleBoostItems.addAll(Arrays.asList(new Item[] {Item.FAMILY_PHOTO, Item.HARMONICA, Item.JOURNAL}));
+    for (Item i : moraleBoostItems ) {
+      if (this.items.contains(i)) {
+        moraleBoostItemsOwn.add(i);
+      }
+    }
+    if (moraleBoostItemsOwn.isEmpty()) {
+      updateMorale(-1);
+      return resultBuilder
+          .message("It is cold and sad here. I know you are lonely, do you want to take some rest?")
+          .morale(-1)
+          .build();
+    }
+    // randomly pick a item from the moralBoostItemsOwn
+    Random rand = new Random();
+    int randomIndex = rand.nextInt(moraleBoostItemsOwn.size());
+    if (moraleBoostItemsOwn.get(randomIndex) == Item.FAMILY_PHOTO ) {
+      updateMorale(3);
+      resultBuilder
+          .message("You found your family photo and it reminds you all the good memories with your family! Your" +
+              " morale is high now!")
+          .morale(3);
+    }
+    else if (moraleBoostItemsOwn.get(randomIndex) == Item.HARMONICA ) {
+      double caloriesBurned = ActivityLevel.LOW.getCaloriesBurned(SuccessRate.LOW);
+      updateWeight(-caloriesBurned);
+      updateMorale(2);
+      resultBuilder
+          .message("You found a harmonica, and you played with it for an hour, your morale is high now!")
+          .morale(2);
+    }
+    else {
+      double caloriesBurned = ActivityLevel.LOW.getCaloriesBurned(SuccessRate.LOW);
+      updateWeight(-caloriesBurned);
+      updateMorale(1);
+      resultBuilder
+          .message(("You found a Journal and a pen, you decide to capture current experience in the journal. " +
+              "Your morale is high now!"))
+          .morale(1);
+    }
+    return resultBuilder.build();
   }
-
+  /**
+   * Activity method for Player to have some rest.
+   *
+   * @return Result of having rest.
+   */
   public Result rest() {
```

Comment:
I think it is sort of cruel that they only burn calories by resting. Maybe we could boost morale a bit? That is unless you are just trying to trick them into resting ;-)